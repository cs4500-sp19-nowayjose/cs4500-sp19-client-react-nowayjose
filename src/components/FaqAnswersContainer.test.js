import React from 'react'
import FAQAnswersContainer from './FaqAnswersContainer'
import TestRenderer from 'react-test-renderer'
import FaqAnswerService from '../services/FAQAnswerService'
const faqAnsService = FaqAnswerService.getInstance()
import setUp from  '../services/FAQAnswerService.mock'


beforeAll(function () {
    setUp()
})

test('FAQAnswersContainer renders correctly', () => {
    setUp()
    const testRenderer = TestRenderer.create(
        <FAQAnswersContainer/>
    )
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})

test('FAQAnswersContainer renders correctly after deleting a faq answer', () => {
    setUp()
    const testRenderer = TestRenderer.create(
        <FAQAnswersContainer/>
    )
    let tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()

    const testInstance = testRenderer.root
    let deleteFaqAns = testInstance.findByProps({className: 'delete'})

    deleteFaqAns.props.onClick({})
    tree = testRenderer.toJSON()
    expect(tree).toMatchSnapshot()
})
