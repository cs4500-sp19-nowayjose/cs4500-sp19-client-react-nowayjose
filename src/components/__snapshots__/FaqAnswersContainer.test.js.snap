// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAQAnswersContainer renders correctly 1`] = `
<div>
  <h3>
    FAQ Answers
  </h3>
  <table
    className="table"
  >
    <thread>
      <tr>
        <th
          scope="col"
        >
          Username
        </th>
        <th
          scope="col"
        >
          Question
        </th>
        <th
          scope="col"
        >
          Answer
        </th>
        <th>
           
        </th>
      </tr>
      <tr>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="username"
            type="text"
            value="Username"
          />
        </th>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="question"
            type="text"
            value="Question"
          />
        </th>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="answer"
            type="text"
            value="New answer"
          />
        </th>
        <th>
           
          <button
            class="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
           
        </th>
        <th>
           
          <button
            class="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Create
          </button>
           
        </th>
      </tr>
    </thread>
    <tbody />
  </table>
</div>
`;

exports[`FAQAnswersContainer renders correctly 2`] = `
<div>
  <h3>
    FAQ Answers
  </h3>
  <table
    className="table"
  >
    <thread>
      <tr>
        <th
          scope="col"
        >
          Username
        </th>
        <th
          scope="col"
        >
          Question
        </th>
        <th
          scope="col"
        >
          Answer
        </th>
        <th>
           
        </th>
      </tr>
      <tr>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="username"
            type="text"
            value="Username"
          />
        </th>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="question"
            type="text"
            value="Question"
          />
        </th>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="answer"
            type="text"
            value="New answer"
          />
        </th>
        <th>
           
          <button
            class="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
           
        </th>
        <th>
           
          <button
            class="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Create
          </button>
           
        </th>
      </tr>
    </thread>
    <tbody />
  </table>
</div>
`;

exports[`FAQAnswersContainer renders correctly after deleting a faq answer 1`] = `
<div>
  <h3>
    FAQ Answers
  </h3>
  <table
    className="table"
  >
    <thread>
      <tr>
        <th
          scope="col"
        >
          Username
        </th>
        <th
          scope="col"
        >
          Question
        </th>
        <th
          scope="col"
        >
          Answer
        </th>
        <th>
           
        </th>
      </tr>
      <tr>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="username"
            type="text"
            value="Username"
          />
        </th>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="question"
            type="text"
            value="Question"
          />
        </th>
        <th>
           
          <input
            onChange={[Function]}
            placeholder="answer"
            type="text"
            value="New answer"
          />
        </th>
        <th>
           
          <button
            class="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Save
          </button>
           
        </th>
        <th>
           
          <button
            class="btn btn-primary btn-block"
            onClick={[Function]}
            type="button"
          >
            Create
          </button>
           
        </th>
      </tr>
    </thread>
    <tbody />
  </table>
</div>
`;
